### PR TITLE
Add obstruction check to queen_valid_move? method and tests

### PR DESCRIPTION
**Pull request by nnadkarny**
- Added an obstruction test to queen_spec.rb
- Added an obstruction check to queen_valid_move? method in queen.rb

### CODE REVIEW

File: spec/models/queen_spec.rb [hunk 1]
```diff
@@ -34,10 +34,22 @@
...
+        it { is_expected.to eq(false) }
+      end
 
-      it { is_expected.to eq(false) }
+      context 'when obstructed' do
+        let(:destination_x) { 6 }
+        let(:destination_y) { 3 }
+
+        it 'should return false' do
+          FactoryGirl.create(:bishop, color: "White", x_pos: 5, y_pos: 2, game_id: game.id)
```
Comment (kirbygit) [inline spec/models/queen_spec.rb]: - use let

**Comment by kirbygit** [comment]
- refactored queen test suite
after the checks all good to go 👍🏻

**Comment by nnadkarny** [comment]
Awesome, thanks @kirbygit!

**Comment by nnadkarny** [comment]
@kirbygit mind doing an official review?  Then we can merge.

**Comment by kirbygit** [comment]
👍🏻

**Pull request merged**